### PR TITLE
Improve DateUtils test coverage

### LINKED ISSUE #311: Fix failing DateUtil tests
The following tests are failing with `mvn clean install`:
- [ERROR]   DateUtilsTest.testCurrentWeekNumberOfYearAndWeeksLeftInYear:57 expected: <1> but was: <43>
- [ERROR]   DateUtilsTest.testWeeksLeftInYearFromCurrentWeek:65 expected: <51> but was: <0>

### LINKED ISSUE #535: Add more DateUtil tests
Add tests for:
- calculateWeeksLeftInYearFromCurrentWeek
- dateIsInCurrentYear()

Edge cases should also be handled

### PR DESCRIPTION
**Pull request by loganrouleau**
## Summary of change
Added testing for new functionality in DateUtils. Fixed the static mocking issue in #311 on the way.

## Related issue

Closes #535 
Closes #311

## Pull request checklist

Please keep this checklist in & ensure you have done the following:

- [x] Read, understood and adhered to our [contributing document](https://github.com/knjk04/book-project/blob/master/CONTRIBUTING.md).
  - [x] Ensure that you were first assigned to a relevant issue before creating this pull request
  - [x] Ensure code changes pass all tests

- [x] Read, understood and adhered to our [style guide](https://github.com/knjk04/book-project/blob/master/STYLEGUIDE.md). A lot of our code reviews are spent on ensuring compliance with our style guide, so it would save a lot of time if this was adhered to from the outset. 

- [x] Filled in the summary, context (if applicable) and related issue section. Replace the square brackets and its placeholder content with your contents. For an example, see any merged in pull request
  - [x] Included a screenshot(s) if a UI change was involved (it may look different on a reviewer's device)

- [x] Created a branch that has a descriptive name (what your branch is for in a few words and includes the issue number at the end, e.g. `test-reading-goal-123`

- [x] Set this pull request to 'draft' if you are still working on it

- [x] Resolved any merge conflicts

For any of the optional checkboxes (e.g. the screenshots one), still check it if it does not apply.

If in doubt, get in touch with us via our [Slack workspace](https://join.slack.com/t/teambookproject/shared_invite/zt-i7lept44-rgJ9yB0A2vedJTLyyfkjKQ)

### CODE REVIEW

**Comment by knjk04** [comment]
Thanks for working on this

> I can reproduce Karan's issue here on Windows with mvn -Dtest=ReadingGoalCalculatorTest,DateUtilsTest test
The issue seems to be statically mocking DateUtils from 2 different classes. The mocking survives as long as the test thread, and I don't think we have any surefire parallelization enabled so every test is on 1 thread. I was able to fix it by assigning Mockito.mockStatic(DateUtils.class) to a variable like private static MockedStatic<DateUtils> mockedDateUtils;, and then unsetting the mock at the end of each test class:

Would it be possible to add your fix to #311 here? Then, it should be possible to also enable the tests you've disabled?

**Comment by loganrouleau** [comment]
Sure, I can update the pull request later today. Then both tickets are taken care of.

File: src/test/java/com/karankumar/bookproject/backend/util/DateUtilsTest.java [hunk 1]
```diff
@@ -78,6 +87,28 @@ void correctlyGetWeeksLeftInYearFromCurrentWeek() {
...
+    }
+
+    @Test
+    void correctlyCheckIfDateIsInFuture() {
+        assertThat(DateUtils.isDateInFuture(MOCK_CURRENT_DATE.plusYears(1))).isTrue();
+        assertThat(DateUtils.isDateInFuture(MOCK_CURRENT_DATE.plusDays(1))).isTrue();
+
+        assertThat(DateUtils.isDateInFuture(MOCK_CURRENT_DATE)).isFalse();
+        assertThat(DateUtils.isDateInFuture(MOCK_CURRENT_DATE.minusDays(1))).isFalse();
+        assertThat(DateUtils.isDateInFuture(MOCK_CURRENT_DATE.minusMonths(1))).isFalse();
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/util/DateUtilsTest.java:109]: Could you please use `assertSoftly()` here?

**Review by knjk04 — CHANGES_REQUESTED** [review]
Thanks for fixing this and adding more tests!

I've made a suggestion to use `assertSoftly()`, so that we can see if other tests fail if an earlier one fails

File: src/test/java/com/karankumar/bookproject/backend/util/DateUtilsTest.java (hunk 1 above)
Comment (loganrouleau) [reply src/test/java/com/karankumar/bookproject/backend/util/DateUtilsTest.java]: Good idea, updated.

**Review by knjk04 — APPROVED** [review]
Thanks a lot for working on this!
